sdk/communication/Azure.Communication.Calling.Server/src/CallClient.cs

```diff
@@ -0,0 +1,604 @@
+            }
+            catch (Exception ex)
+            {
+                scope.Failed(ex);
+                throw;
+            }
+        }
+
+        /// <summary> Remove a participants from the call. </summary>
+        /// <param name="callLegId"> The call leg id. </param>
+        /// <param name="participantId"> The participant id. </param>
+        /// <param name="cancellationToken"> The cancellation token to use. </param>
+        /// <exception cref="RequestFailedException">The server returned an error. See <see cref="Exception.Message"/> for details returned from the server.</exception>
+        /// <exception cref="ArgumentNullException"> <paramref name="callLegId"/> or <paramref name="participantId"/> is null. </exception>
+        public virtual async Task<Response> RemoveParticipantAsync(string callLegId, string participantId, CancellationToken cancellationToken = default)
+        {
+            using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(CallClient)}.{nameof(RemoveParticipant)}");
+            scope.Start();
+            try
+            {
+                Argument.AssertNotNullOrEmpty(callLegId, nameof(callLegId));
+                Argument.AssertNotNullOrEmpty(participantId, nameof(participantId));
+
+                return await RestClient.RemoveParticipantAsync(callLegId, participantId, cancellationToken).ConfigureAwait(false);
+            }
+            catch (Exception ex)
+            {
+                scope.Failed(ex);
+                throw;
+            }
+        }
+
+        /// <summary> Remove a participants from the call. </summary>
+        /// <param name="callLegId"> The call leg id. </param>
+        /// <param name="participantId"> The participant id. </param>
+        /// <param name="cancellationToken"> The cancellation token to use. </param>
+        /// <exception cref="RequestFailedException">The server returned an error. See <see cref="Exception.Message"/> for details returned from the server.</exception>
+        /// <exception cref="ArgumentNullException"> <paramref name="callLegId"/> or <paramref name="participantId"/> is null. </exception>
+        public virtual Response RemoveParticipant(string callLegId, string participantId, CancellationToken cancellationToken = default)
+        {
+            using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(CallClient)}.{nameof(RemoveParticipant)}");
+            scope.Start();
+            try
+            {
+                Argument.AssertNotNullOrEmpty(callLegId, nameof(callLegId));
+                Argument.AssertNotNullOrEmpty(participantId, nameof(participantId));
+
+                return RestClient.RemoveParticipant(callLegId, participantId, cancellationToken);
+            }
+            catch (Exception ex)
+            {
+                scope.Failed(ex);
+                throw;
+            }
+        }
+
+        private static T AssertNotNull<T>(T argument, string argumentName)
```

Comment:
I dont think you need your own implementation of these asserts. Please directly use the Argument class from Core if you need to perfrom assertNotNull/AssertNotNullOrEmpty checks

Reply:
We use it for constructor parameter validation so that the constructors are decent and clean.
The [Sms client ](https://github.com/Azure/azure-sdk-for-net/blob/557ebceca38ad031043645f844d5b8b30f3cb338/sdk/communication/Azure.Communication.Sms/src/SmsClient.cs#L32
)has implemented this as well.

Reply:
I see, could we actually move this into Argument class in Core instead? It's not ideal having these checks being copied repeatedly in the clients

Reply:
https://github.com/Azure/azure-sdk-for-net/pull/21094#discussion_r633861192